testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -905,29 +896,59 @@ impl RelayerThread {
                 })?
         };
 
-        let won_last_sortition = last_winner_snapshot.miner_pk_hash == Some(mining_pkh);
+        let won_last_sortition = last_block_election_snapshot.miner_pk_hash == Some(mining_pkh);
         debug!(
             "Relayer: Current burn block had no sortition. Checking for tenure continuation.";
             "won_last_sortition" => won_last_sortition,
             "current_mining_pkh" => %mining_pkh,
-            "last_winner_snapshot.miner_pk_hash" => ?last_winner_snapshot.miner_pk_hash,
+            "last_block_election_snapshot.consensus_hash" => %last_block_election_snapshot.consensus_hash,
+            "last_block_election_snapshot.miner_pk_hash" => ?last_block_election_snapshot.miner_pk_hash,
             "canonical_stacks_tip_id" => %canonical_stacks_tip,
             "canonical_stacks_tip_ch" => %canonical_stacks_tip_ch,
-            "block_election_ch" => %block_election_snapshot.consensus_hash,
             "burn_view_ch" => %new_burn_view,
         );
 
         if !won_last_sortition {
             return Ok(());
         }
 
+        let last_non_empty_sortition_snapshot =
+            SortitionDB::get_block_snapshot_consensus(self.sortdb.conn(), &canonical_stacks_tip_ch)
+                .map_err(|e| {
+                    error!("Relayer: failed to get block snapshot for canonical tip: {e:?}");
+                    NakamotoNodeError::SnapshotNotFoundForChainTip
+                })?
+                .ok_or_else(|| {
+                    error!("Relayer: failed to get block snapshot for canonical tip");
+                    NakamotoNodeError::SnapshotNotFoundForChainTip
+                })?;
+
+        let won_last_non_empty_sortition_snapshot =
+            last_non_empty_sortition_snapshot.miner_pk_hash == Some(mining_pkh);
+
+        let (parent_tenure_start, block_election_snapshot, reason) =
+            if !won_last_non_empty_sortition_snapshot {
+                debug!("Relayer: Failed to issue a tenure change payload in our last tenure. Issue a new tenure change payload.");
+                (
+                    canonical_stacks_tip,
```

Comment:
I don't actually know if this is correct...is the parent_tenure_start the same as the canonical_stacks_tip in a tenure change ?

Reply:
Not always. The miner's block commit transaction includes the parent tenure start (in 2.x, this was the block hash of the actual block they are submitting, but in nakamoto it is the parent tenure's start block ID).

This should be the value in `last_block_election_snapshot.winning_stacks_block_hash` -- which you'll have to "cast" to a StacksBlockId:

```rust
StacksBlockId(last_block_election_snapshot.winning_stacks_block_hash.0)
```

Reply:
THanks! It did seem def sus since we had a specific comment about tenure extends being off the chain tip.

Will fix!